src/indexing.jl

```diff
@@ -369,4 +369,11 @@ Base.checkindex(B::Type{Bool}, inds::AbstractUnitRange, i::StaticIndexing{T}) wh
 
 # unsafe_view
 
-Base.unsafe_view(A::AbstractArray, i::StaticIndexing{T}) where T = Base.unsafe_view(A, unwrap(i))
+unwrap_if_needed(i::StaticIndexing) = unwrap(i)
+unwrap_if_needed(i::Base.ViewIndex) = i
+
+Base.unsafe_view(A::AbstractArray, is::Union{StaticIndexing,Base.ViewIndex}...) = Base.unsafe_view(A, map(unwrap_if_needed, is)...)
```

Comment:
This method signature does make me uneasy because it can also match cases where none of the argument types are defined in this package.

However it seems the method specificity rules make it mostly work (or at least I couldn't find a case where there's an ambiguity).

I did find the following bug though in the process:

```julia
julia> a = rand(3,4,5,6);

julia> b = @view a[:,:,1,1]
3×4 view(::Array{Float64,4}, :, :, 1, 1) with eltype Float64:
 0.450313  0.311055  0.747741  0.966306
 0.984626  0.313968  0.798511  0.71977 
 0.486325  0.42368   0.743077  0.285142

julia> @view b[SA[1,2],1]
2-element view(::Array{Float64,4}, StaticArrays.StaticIndexing{SArray{Tuple{2},Int64,1,2}}([1, 2]), StaticArrays.StaticIndexing{Int64}(1), StaticArrays.StaticIndexing{Int64}(1), StaticArrays.StaticIndexing{Int64}(1)) with eltype Float64 with indices SOneTo(2):
#undef

```

Reply:
Thanks for looking at it :slightly_smiling_face: . I've already discovered this problem but I don't know yet how to fix it. I'm working on it.